File: output/CodeGen/codenet/C++/Java/s189738089.java
public class s189738089 {
    public int[] findMinHeightTrees(int n, int[] height) {
        int[] dp = new int[n + 1];
        dp[0] = 0;
        for (int i = 1; i <= n; i++) {
            int min = INF;
            for (int j = 1; j <= i; j++) {
                if (height[j - 1] < min) {
                    min = height[j - 1];
                }
            }
            dp[i] = min;
        }
        return dp;
    }
}

File: output/StarCoder/codenet/C++/Python/s189738089.py
import sys

def main():
    # input
    N, K = map(int, sys.stdin.readline().split())
    h = list(map(int, sys.stdin.readline().split()))

    # dp
    dp = [float('inf') for _ in range(101000)]
    dp[0] = 0
    for i in range(N):
        for j in range(1, K + 1):
            dp[i + j] = min(dp[i + j], dp[i] + abs(h[i] - h[i + j]))

    # output
    print(dp[N - 1])

if __name__ == '__main__':
    main()